sycl/include/CL/sycl/builtins.hpp

```diff
@@ -27,6 +27,10 @@ namespace __sycl_std = __host_std;
 } // namespace sycl
 } // namespace cl
 
+#ifdef __SYCL_DEVICE_ONLY__
+extern int __spirv_ocl_printf(const char *__format, ...);
```

Comment:
This declares a `format` string in generic address space, or it will be addrspacecasted to generic from private/global/local.

This is a problem, because OpenCL specification expects the format string to be in *constant* address space, and it  means that a format string has to be a constant (global) string in LLVM IR - it cannot be allocated on a stack, passed as an argument or computed at runtime. OpenCL CPU backend relies on this fact to compute total size of arguments passed to a printf call.

Reply:
Updated declaration, added a bunch of disclaimers